docs/user-guide/v3_migration.rst

```diff
@@ -0,0 +1,226 @@
+3.0 Migration Guide
+===================
+
+Zarr-Python 3 represents a major refactor of the Zarr-Python codebase. Some of the
+goals motivating this refactor included:
+
+* adding support for the Zarr V3 specification (along with the Zarr V2 specification)
+* cleaning up internal and user facing APIs
+* improving performance (particularly in high latency storage environments like
+  cloud object store)
+
+To accommodate this, Zarr-Python 3.0 introduces a number of changes to the API, including a number
+of significant breaking changes and pending deprecations.
+
+This page provides a guide highlighting the most notable changes to help you
+migrate your code from version 2 to version 3. If we have missed anything, please
+open a `GitHub issue <https://github.com/zarr-developers/zarr-python/issues/new>`_
+so we can improve this guide.
+
+Compatibility target
+--------------------
+
+The goals described above necessitated some breaking changes to the API (hence the
+major version update), but where possible we have maintained backwards compatibility
+in the most widely used parts of the API. This in the :class:`zarr.Array` and
+:class:`zarr.Group` classes and the "top-level API" (e.g. :func:`zarr.open_array` and
+:func:`zarr.open_group`).
```

Comment:
```suggestion
The goals described above necessitated some breaking changes to the API (hence the
major version update), but where possible we have maintained backwards compatibility
with functions in the top-level ``zarr`` namespace (e.g. :func:`zarr.open_array` and :func:`zarr.open_group`).
```

Reply:
We have also attempted to maintain backward compatibility for the Group/Array classes. Neither the top level API or the Array/Group classes hit 100% backward compatibility.